oxd-server/src/main/java/org/gluu/oxd/server/service/StateService.java

```diff
@@ -51,12 +51,21 @@ public void deleteExpiredObjectsByKey(String key) {
     }
 
     public String putState(String state) {
-        persistenceService.createExpiredObject(new ExpiredObject(state, ExpiredObjectType.STATE, configurationService.get().getStateExpirationInMinutes()));
+        persistenceService.createExpiredObject(new ExpiredObject(state, state, ExpiredObjectType.STATE, configurationService.get().getStateExpirationInMinutes()));
         return state;
     }
 
     public String putNonce(String nonce) {
-        persistenceService.createExpiredObject(new ExpiredObject(nonce, ExpiredObjectType.NONCE, configurationService.get().getNonceExpirationInMinutes()));
+        persistenceService.createExpiredObject(new ExpiredObject(nonce, nonce, ExpiredObjectType.NONCE, configurationService.get().getNonceExpirationInMinutes()));
         return nonce;
     }
+
+    public String putRequestObject(String oxdId, String requestObject) {
```

Comment:
Introduce `RequestObjectService` class and lets keep request object related functionality there.

Reply:
Done